src/promptflow/promptflow/batch/_base_executor_proxy.py

```diff
@@ -22,6 +22,15 @@
 
 
 class AbstractExecutorProxy:
+    @classmethod
+    def get_tool_metadata(cls, flow_file: Path, working_dir: Optional[Path] = None) -> dict:
```

Comment:
Why is this method included in `AbstractExecutorProxy`? It appears to be specific to `CSharpExecutorProxy`. 

If it's necessary for all proxies, maybe simply using get_tool_metadata with a pass is enough. Implementing it with a NotImplementedError() seems unnecessary, especially since I believe `PythonExecutorProxy` may not need this method.

Reply:
As Xingzhi is taking marriage leave, coud we address this in future PR? 
I think this needs further discussion. I think get aviable tool meta is essential for all language UI yaml-first authoring experience.